handlers/jaeger.go

```diff
@@ -19,6 +44,12 @@ func GetJaegerInfo(w http.ResponseWriter, r *http.Request) {
 		RespondWithError(w, code, err.Error())
 		return
 	}
+	/** Request the proxy solution */
+	urlKiali, _ := getOpenshiftRouteURL("istio-system", "kiali")
```

Comment:
Mmm perhaps I'm wrong, but how this will work in a kubernetes deployment ?
Should be some conditional here ?

Reply:
I think that works  it's using the kubernetes library
```go
func getOpenshiftRouteURL(namespace string, name string) (string, error) {
	client, err := kubernetes.NewOSRouteClient()
	if err != nil {
		return "", err
	}
	return client.GetRoute(namespace, name)
}
```

Reply:
aaa the namespace , yes we should make some conditional

Reply:
NewOSRouteClient is specific for OpenShift scenarios, so this probably will through an error on kubernetes deployments.

Reply:
So this should be fail in grafana too. we should test it

```go
func GetGrafanaInfo(w http.ResponseWriter, r *http.Request) {
	info, code, err := getGrafanaInfo(getOpenshiftRouteURL, getService)
	if err != nil {
		log.Error(err)
		RespondWithError(w, code, err.Error())
		return
	}
	RespondWithJSON(w, code, info)
}
```

Reply:
getGrafanaInfo tests if you are in OpenShift or not, failing into a config var for kubernetes.

Reply:
Not hardcoding "istio-system" and "kiali" is good but I fail to see how it solved the issue pointed by @lucasponce . I guess it will still fail with plain kubernetes.